qiita_db/metadata_template/util.py

```diff
@@ -132,12 +116,30 @@ def load_template_to_dataframe(fn, strip_whitespace=True, index='sample_name'):
         holdfile.insert(0, "%s\n" % '\t'.join(headers))
         # The QIIME parser fixes the index and removes the #
         index = 'SampleID'
+    else:
```

Comment:
Any reason why these checks are done only if the file is not a mapping file? Note that users can upload a QIIME mapping file instead of sample + prep information, so I think it is better if these checks are always executed (as they're before).